compiler/calls.go

```diff
@@ -38,6 +39,48 @@ func (c *Compiler) createCall(fn llvm.Value, args []llvm.Value, name string) llv
 		fragments := c.expandFormalParam(arg)
 		expanded = append(expanded, fragments...)
 	}
+	fnParams := fn.Type().ElementType().ParamTypes()
+pcheck:
+	if len(fnParams) != len(expanded) {
+		if len(fnParams) < len(expanded) {
+			// too many parameters
+			// attempt to trim off parent handle/etc. if not used
```

Comment:
Is this actually needed right now? If not, I'd recommend just erroring here.

Reply:
Can you give an example?

If this is for functions marked `//export`, then it is incorrect: they use a different ABI and may diverge in the future (but are nearly the same right now except for the last two parameters). `//export` tries to strictly follow the C ABI while TinyGo internally uses a different ABI that may be more efficient (and is not stable).

Reply:
@aykevl As far as I can tell, this only seems to happen when invoking runtime.yield via linkname on the tasks scheduler.

Reply:
Also, the tests pass without this parameter stripping. I would suggest using the correct number of parameters across mismatching calls.

Reply:
Alright. I switched this to a cleaner (somewhat) mechanism.